--- ./CcspLMLite/source/lm_main.c
+++ ./CcspLMLite/source/lm_main.c	2015-06-15 07:21:49.128343225 -0600
@@ -118,14 +118,40 @@
 
 /* It may be updated by different threads at the same time? */
 ULONG HostsUpdateTime = 0;
+
 pthread_mutex_t PollHostMutex;
 pthread_mutex_t LmHostObjectMutex;
 
+int logOnlineDevicesCount()
+{
+	PLmObjectHost   pHost      = NULL;
+	int NumOfOnlineDevices = 0;
+	int i;
+	for ( i = 0; i < lmHosts.numHost; i++ )
+	{               
+		pHost = lmHosts.hostArray[i];
+
+		if(pHost->bBoolParaValue[LM_HOST_ActiveId])
+		{
+			NumOfOnlineDevices ++;
+		}
+	}
+	CcspWifiTrace(("RDK_LOG_WARN, CONNECTED_CLIENTS_COUNT : %d \n",NumOfOnlineDevices));
+}
+
 #define LM_SET_ACTIVE_STATE_TIME(x, y) LM_SET_ACTIVE_STATE_TIME_(__LINE__, x, y)
 inline void LM_SET_ACTIVE_STATE_TIME_(int line, LmObjectHost *pHost,BOOL state){
     if(pHost->bBoolParaValue[LM_HOST_ActiveId] != state){
+		if((strstr(pHost->pStringParaValue[LM_HOST_Layer1InterfaceId],"WiFi"))) {
+			if(state) {
+				CcspWifiTrace(("RDK_LOG_WARN, Wifi client mac-%s appeared online \n",pHost->pStringParaValue[LM_HOST_PhysAddressId]));
+			}  else {
+				CcspWifiTrace(("RDK_LOG_WARN, Wifi client mac-%s gone offline \n",pHost->pStringParaValue[LM_HOST_PhysAddressId]));
+			}
+		}
         pHost->bBoolParaValue[LM_HOST_ActiveId] = state;
         pHost->activityChangeTime = time((time_t*)NULL);
+		logOnlineDevicesCount();
 		PRINTD("%d: mac %s, state %d time %d\n",line ,pHost->pStringParaValue[LM_HOST_PhysAddressId], state, pHost->activityChangeTime);
     }
 } 
@@ -301,6 +327,7 @@
 
     if(!physAddress || \
        0 == strcmp(physAddress, "00:00:00:00:00:00")) return NULL;
+
     if(strlen(physAddress) != MACADDR_SZ-1) return NULL;
     PLmObjectHost pHost = Hosts_FindHostByPhysAddress(physAddress);
     if(pHost) return pHost;
@@ -1015,6 +1042,7 @@
                     }
                     else if ( pHost && pHost->l1unReachableCnt == 0 )
                     {
+
                         if ( hosts[i].status == LM_NEIGHBOR_STATE_REACHABLE )
                         {
                             LM_SET_ACTIVE_STATE_TIME(pHost, TRUE);
@@ -1070,7 +1098,6 @@
 {
     pthread_mutex_lock(&PollHostMutex);
     Hosts_SyncArp();
-
     Hosts_SyncDHCP();
     Hosts_SyncWifi();
     Hosts_SyncMoCA();
